src/vfio_device.rs

```diff
@@ -177,7 +178,9 @@ impl VfioContainer {
     ///
     /// # Arguments
     /// * `device_fd`: file handle of the VFIO device.
-    pub fn new(device_fd: Arc<DeviceFd>) -> Result<Self> {
+    pub fn new(
+        #[cfg(any(feature = "kvm", feature = "mshv"))] device_fd: Arc<DeviceFd>,
```

Comment:
Have you thought about changing `device_fd` to `Option<Arc<DeviceFd>>`? I guess that still requires the code to depend on `DeviceFd`'s visibility (i.e. can't avoid the dependencies on KVM or MSHV crate), so that's not very ideal?

Reply:
There's another solution in #44, which solve the same problem.

Reply:
Thanks Gerry. I will have a look.